elements/pl-python-variable/pl-python-variable.py

```diff
@@ -1,52 +1,119 @@
 import prairielearn as pl
 import lxml.html
+import pandas
+from typing import cast, List, Any
 
 NO_HIGHLIGHT_DEFAULT = False
-PREFIX_DEFAULT = ''
-SUFFIX_DEFAULT = ''
+PREFIX_DEFAULT = ""
+SUFFIX_DEFAULT = ""
 TEXT_DEFAULT = False
 SHOW_HEADER_DEFAULT = True
 SHOW_INDEX_DEFAULT = True
 SHOW_DIMENSIONS_DEFAULT = True
+SHOW_DATATYPE_DEFAULT = False
+ADD_LINE_BREAKS_DEFAULT = False
 
 
-def prepare(element_html, data):
+def prepare(element_html: str, data: pl.QuestionData) -> None:
     element = lxml.html.fragment_fromstring(element_html)
-    pl.check_attribs(element, required_attribs=['params-name'], optional_attribs=['text', 'no-highlight', 'prefix', 'suffix', 'show-header', 'show-index', 'show-dimensions'])
+    pl.check_attribs(
+        element,
+        required_attribs=["params-name"],
+        optional_attribs=[
+            "text",
+            "no-highlight",
+            "prefix",
+            "suffix",
+            "show-header",
+            "show-index",
+            "show-dimensions",
+            "add-line-breaks",
+            "show-dtype",
+        ],
+    )
 
 
-def render(element_html, data):
+def render(element_html: str, data: pl.QuestionData) -> str:
     element = lxml.html.fragment_fromstring(element_html)
-    force_text = pl.get_boolean_attrib(element, 'text', TEXT_DEFAULT)
-    varname = pl.get_string_attrib(element, 'params-name')
-    show_header = pl.get_boolean_attrib(element, 'show-header', SHOW_HEADER_DEFAULT)
-    show_index = pl.get_boolean_attrib(element, 'show-index', SHOW_INDEX_DEFAULT)
-    show_dimensions = pl.get_boolean_attrib(element, 'show-dimensions', SHOW_DIMENSIONS_DEFAULT)
+    force_text = pl.get_boolean_attrib(element, "text", TEXT_DEFAULT)
+    varname = pl.get_string_attrib(element, "params-name")
+    show_header = pl.get_boolean_attrib(element, "show-header", SHOW_HEADER_DEFAULT)
+    show_index = pl.get_boolean_attrib(element, "show-index", SHOW_INDEX_DEFAULT)
+    show_dimensions = pl.get_boolean_attrib(
+        element, "show-dimensions", SHOW_DIMENSIONS_DEFAULT
+    )
+    show_dtype = pl.get_boolean_attrib(element, "show-dtype", SHOW_DATATYPE_DEFAULT)
+    add_line_breaks = pl.get_boolean_attrib(
+        element, "add-line-breaks", ADD_LINE_BREAKS_DEFAULT
+    )
 
-    if varname not in data['params']:
-        raise Exception('Could not find {} in params!'.format(varname))
+    if varname not in data["params"]:
+        raise ValueError(f"Could not find {varname} in params!")
```

Comment:
`KeyError` is probably more appropriate here.

```suggestion
        raise KeyError(f"Could not find {varname} in params!")
```